core/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -0,0 +1,144 @@
+        this.oldIndex = oldIndex;
+        this.newIndex = newIndex;
+        this.dryRun = dryRun;
+        this.rolledOver = rolledOver;
+        this.conditionStatus = conditionResults.stream()
+            .map(result -> new AbstractMap.SimpleEntry<>(result.condition.toString(), result.matched))
+            .collect(Collectors.toSet());
+    }
+
+    /**
+     * Returns the name of the index that the request alias was pointing to
+     */
+    public String getOldIndex() {
+        return oldIndex;
+    }
+
+    /**
+     * Returns the name of the index that the request alias currently points to
+     */
+    public String getNewIndex() {
+        return newIndex;
+    }
+
+    /**
+     * Returns the statuses of all the request conditions
+     */
+    public Set<Map.Entry<String, Boolean>> getConditionStatus() {
+        return conditionStatus;
+    }
+
+    /**
+     * Returns if the rollover execution was skipped even when conditions were met
+     */
+    public boolean isDryRun() {
+        return dryRun;
+    }
+
+    /**
+     * Returns if the rollover was not simulated and the conditions were met
+     */
+    public boolean isRolledOver() {
+        return rolledOver;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        oldIndex = in.readString();
+        newIndex = in.readString();
+        int conditionSize = in.readVInt();
+        Set<Map.Entry<String, Boolean>> conditions = new HashSet<>(conditionSize);
+        for (int i = 0; i < conditionSize; i++) {
+            String condition = in.readString();
+            boolean satisfied = in.readBoolean();
+            conditions.add(new AbstractMap.SimpleEntry<>(condition, satisfied));
+        }
+        conditionStatus = conditions;
+        dryRun = in.readBoolean();
+        rolledOver = in.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(oldIndex);
+        out.writeString(newIndex);
+        out.writeVInt(conditionStatus.size());
+        for (Map.Entry<String, Boolean> entry : conditionStatus) {
+            out.writeString(entry.getKey());
+            out.writeBoolean(entry.getValue());
+        }
+        out.writeBoolean(dryRun);
+        out.writeBoolean(rolledOver);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(Fields.OLD_INDEX, oldIndex);
+        builder.field(Fields.NEW_INDEX, newIndex);
+        builder.field(Fields.ROLLED_OVER, rolledOver);
+        builder.field(Fields.DRY_RUN, dryRun);
+        builder.startObject(Fields.CONDITIONS);
+        for (Map.Entry<String, Boolean> entry : conditionStatus) {
+            builder.field(entry.getKey(), entry.getValue());
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    static final class Fields {
```

Comment:
I think this class is unnecessary, just make the constants part of the outer class